packages/frontend/src/components/common/SpaceActionModal/CreateSpaceSelectAccessType.tsx

```diff
@@ -0,0 +1,76 @@
+import { Icon } from '@blueprintjs/core';
+import { Select2 } from '@blueprintjs/select';
+import { FC } from 'react';
+import { useParams } from 'react-router-dom';
+import { useProject } from '../../../hooks/useProject';
+import {
+    AccessRole,
+    ChangeAccessButton,
+    FlexWrapper,
+    PrimaryAndSecondaryTextWrapper,
+    PrimaryText,
+    SecondaryText,
+    ShareCircle,
+    UserRole,
+} from '../ShareSpaceModal/ShareSpaceModal.style';
+import {
+    AccessOption,
+    renderAccess,
+    SpaceAccessOptions,
+    SpaceAccessType,
+} from '../ShareSpaceModal/ShareSpaceSelect';
+
+interface ShareSpaceAccessTypeProps {
+    selectedAccess: AccessOption;
+    setSelectedAccess: (access: AccessOption) => void;
+}
+
+export const CreateSpaceSelectAccessType: FC<ShareSpaceAccessTypeProps> = ({
+    selectedAccess,
+    setSelectedAccess,
+}) => {
+    const { projectUuid } = useParams<{ projectUuid: string }>();
```

Comment:
for components that we put in common I recommend having a required prop of `projectUuid: string` instead of getting params from the url. I had a case with project settings when I was moving some parts around and broke half of the functionality because they relied to the URL.